### PR TITLE
Challenge 141-160

### PR DESCRIPTION
**Pull request by Joose1992**
-Added tags according with the difficulty of the exercise. -Added description tags.

### CODE REVIEW

**Comment by kevmoens** [comment]
A tag is a key value pair.  I suggest reviewing you tags as the key should be for grouping and the value should be the detail. 
Example is 
"Array" : "Mean"
I would suggest changing it to two pairs like the following:
"Math" : "Mean"
"Array" : "Iterating"

File: Ellabit/Challenges/Challenge143Returnthefourletterstrings.cs [hunk 1]
```diff
@@ -84,9 +84,9 @@ public class TestChallenge
 
 notes you can expect valid strings for all test cases.""";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { {"String", "Iteration"}, {"Array", "String Iteration"}, { "Level", "2"} };
         public Dictionary<string, string> Tags { get => _tags; set => _tags = value; }
-        public bool ShowBlockly { get => false; }
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge143Returnthefourletterstrings.cs:89]: IronBlock's Blockly doesn't have a way to return an array.  Thus I would set Blockly = false

File: Ellabit/Challenges/Challenge147Makeacirclewithoop.cs [hunk 2]
```diff
@@ -89,9 +89,9 @@ public class TestChallenge
 Notes
 Don't worry about floating point precision - I've factored in the chance that your answer may be more or less accurate than mine. This is more of a tutorial than a challenge so the topic covered may be considered advanced, yet the challenge is more simple - so if this challenge gets labelled as easy, don't worry too much.""";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Math", "Geometry"}, { "Level", "2"} };
         public Dictionary<string, string> Tags { get => _tags; set => _tags = value; }
-        public bool ShowBlockly { get => false; }
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge147Makeacirclewithoop.cs:94]: IronBlock's Blockly doesn't support classes.  blockly = false

File: Ellabit/Challenges/Challenge151Removeduplicatesfromanarray.cs [hunk 3]
```diff
@@ -81,9 +81,9 @@ public class TestChallenge
 
 notes tests contain arrays with""";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Array", "Duplicates"}, { "Level","2"} };
         public Dictionary<string, string> Tags { get => _tags; set => _tags = value; }
-        public bool ShowBlockly { get => false; }
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge151Removeduplicatesfromanarray.cs:86]: IronBlock's Blockly doesn't have a way to return an array. Thus I would set Blockly = false

**Comment by kevmoens** [comment]
Thank you for updating ShowBlockly in all these places.  Although, I also noted a couple places where ShowBlockly should stay false in File Changes.

**Comment by Joose1992** [comment]
Got it, I will work on that, thank you for the review.

**Pull request merged**